File: 2nd round/locate-the-fulcrum/locate-the-fulcrum.java
import java.util.*;
public class Solution{
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int t =sc.nextInt();
        while(t>0){
            t--;
            int n=sc.nextInt();
            int i=n*n;
            i+=n;
            //n+=i;
            i/=2;
            n=(int)(Math.sqrt(i));
            if(n*n==i){
                System.out.println("Yes");
            }
            else{
                System.out.println("No");            }
        }
    }
}

File: 2nd round/locate-the-fulcrum/locate-the-fulcrum.py
import math
t=int(input())
while(t):
    t-=1
    n=int(input())
    i=n*n
    n+=i
    n//=2
    i=int(math.sqrt(n))
    if(i*i==n):
        print("Yes")
    else:
        print("No")
